### ISSUE TITLE
Stem is currently unmaintained. Maybe Onionshare wants to step up and take care of it?

### ISSUE THREAD

**Issue by nodens** [onionshare/onionshare#1634]
Hi,

While preparing the (long overdue) Debian packaging of later versions of Onionshare, we stumbled over the stem issue.
We asked the Tor Project to release their last fixes, so it could be updated in Debian (rather than packaging your fork).

They did, however they also clearly stated that [stem is currently unmaintained](https://lists.torproject.org/pipermail/tor-dev/2022-September/014779.html), although not deprecated, since there is no easy replacement right now.

Question is, would you take over the maintenance? As stated in the message linked above, they would hand over the maintenance if someone is "serious about it". What do you think?

Thanks!

**Comment by emdee-is** [comment]
It looks like he has: https://github.com/onionshare/cepa
I'm moving my code to cepa to test it out.

@micahflee Do you want to swap stem for cepa in onionshare?

It looks like the stem repo has been permanently abandoned and  left in a mess that can't be sorted out: https://github.com/torproject/stem/issues/123

Distros and apps like nyx are now lagging about a year now and it would be good to move on.

**Comment by mig5** [comment]
We forked to cepa for the Client Auth support that we contributed upstream but which had not yet made its way to a Stem release.

Stem has since released several new versions including that fix, so we have effectively abandoned 'cepa' and back to using Stem.

I can see that Stem seems to have quite active commits happening (just 5 days ago), presumably someone is back to maintaining it (or nearly) - I also note atagar, the original developer, is one of the people who seems to be committing to it again.
